File: BiosenseSignal_Flutter_SDK_5.6.3/ios/Classes/BiosenseSignalEventChannel.swift
import Foundation


class BiosenseSignalEventChannel: NSObject, FlutterStreamHandler {
    
    static let eventChannelId = "plugins.biosensesignal.com/sdk_events"
    private var eventSink: FlutterEventSink?
    
    init(messenger: FlutterBinaryMessenger) {
        super.init()
        let channel = FlutterEventChannel(name: BiosenseSignalEventChannel.eventChannelId, binaryMessenger: messenger)
        channel.setStreamHandler(self)
    }
    
    func sendEvent(name: String, payload: Any) {
        DispatchQueue.main.async {
            self.eventSink?([
                "event": name,
                "payload": payload
            ])
        }
    }
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}

File: BiosenseSignal_Flutter_SDK_5.6.3/android/src/main/kotlin/com/biosensesignal/flutter_sdk/BiosenseSignalEventChannel.kt
package com.biosensesignal.flutter_sdk

import io.flutter.plugin.common.EventChannel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class BiosenseSignalEventChannel(
    eventChannel: EventChannel
): EventChannel.StreamHandler {

    private var eventSink: EventChannel.EventSink? = null
    private val mainScope = CoroutineScope(Dispatchers.Main)

    init {
        eventChannel.setStreamHandler(this)
    }

    fun sendEvent(name: String, payload: Any) {
        eventSink?.let { sink ->
            mainScope.launch {
                sink.success(mapOf(
                    Pair("event", name),
                    Pair("payload", payload)
                ))
            }
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

}